src/interpreter/mod.rs

```diff
@@ -1,8 +1,12 @@
 //! WebAssembly interpreter module.
 
-/// Interpreter error.
+/// Custom user error.
+pub trait CustomUserError: 'static + ::std::fmt::Display + ::std::fmt::Debug + Clone + PartialEq {
```

Comment:
maybe call it UserError?

I'm worried about a lot entries in public api are named as SomethingSomethingThing